### PR TITLE
Use css variables for page size values RSC-506 

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-506

I created CSS variables to replace the SCSS variables used for page layout.

In doing so I learned that CSS variables are emitting code, so to get around the repetition problem I moved the CSS variables into the `_page-layout.scss` file.

I also replaced instances of the SCSS variables with the appropriate CSS variables.

### CODE REVIEW

File: lib/src/base-styles/_nx-page-layout.scss [hunk 1]
```diff
@@ -8,6 +8,13 @@
 /*
   #Layout
 */
+:root {
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-page-layout.scss:11]: We already have a section for these at the bottom of `_nx-globals.scss`

File: lib/src/base-styles/_nx-page-layout.scss [hunk 2]
```diff
@@ -8,6 +8,13 @@
...
+  --page-width-max: 1600px;
+  --page-width-min: 1366px;
+  --content-width: 76%;
+  --sidebar-width: 100% - var(--content-width);
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-page-layout.scss:15]: I'm surprised this doesn't need to be a calc

File: lib/src/base-styles/_nx-page-layout.scss [hunk 3]
```diff
@@ -83,7 +90,7 @@
   }
 
   .nx-page {
-    min-width: $nx-page-width-min - $nx-scrollbar-width;
+    min-width: var(--page-width-max) - $nx-scrollbar-width;
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-page-layout.scss:93]: I'd also expect this to need to be a calc

File: lib/src/base-styles/_nx-page-layout.scss (hunk 1 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-page-layout.scss]: > We already have a section for these at the bottom of `_nx-globals.scss`

https://github.com/sonatype/sonatype-react-shared-components/pull/328/commits/9d6f28ce5dadad7cebf9d9e225d93582fed82f95

File: lib/src/base-styles/_nx-page-layout.scss (hunk 2 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-page-layout.scss]: > I'm surprised this doesn't need to be a calc

In this case the units are the same, it's `100% - 76%`.

File: lib/src/base-styles/_nx-page-layout.scss (hunk 3 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-page-layout.scss:86]: > I'd also expect this to need to be a calc

Same as above the units are the same. `1600px - 15px`

Comment (glenhunter) [reply lib/src/base-styles/_nx-page-layout.scss]: FYI I fixed the min/max issue above here: https://github.com/sonatype/sonatype-react-shared-components/pull/328/commits/484bdde0327df578e0d550f1200762378eb06c7b

Comment (rpokorny) [reply lib/src/base-styles/_nx-page-layout.scss]: > Same as above the units are the same. 1600px - 15px

That doesn't matter; CSS only does math in `calc` expressions.  If you try to do math without one that would be SCSS math, which of course can't work with runtime values like a `var()`.  See this screenshot for proof that the `min-width` is not working:

image

Comment (glenhunter) [reply lib/src/base-styles/_nx-page-layout.scss]: > > Same as above the units are the same. 1600px - 15px
> 
> That doesn't matter; CSS only does math in `calc` expressions. If you try to do math without one that would be SCSS math, which of course can't work with runtime values like a `var()`. See this screenshot for proof that the `min-width` is not working:
> 

I came to the same conclusion this weekend but for different reasons, namely what would happen if a consumer set one of the variables to a different unit. So I made the change here: https://github.com/sonatype/sonatype-react-shared-components/pull/328/commits/92071b851ee12d9c3139fe32eb6c39b80aa3c008

Note that I had to move the scrollbar width into a CSS var or the calc for page scrolling wouldn't work in real time.

**Review by rpokorny — APPROVED** [review]

File: lib/src/base-styles/_nx-page-layout.scss (hunk 3 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-page-layout.scss]: > Note that I had to move the scrollbar width into a CSS var or the calc for page scrolling wouldn't work in real time.

That doesn't sound right to me, but I'm not against move that to a CSS variable.

Comment (glenhunter) [reply lib/src/base-styles/_nx-page-layout.scss]: > > Note that I had to move the scrollbar width into a CSS var or the calc for page scrolling wouldn't work in real time.
> 
> That doesn't sound right to me, but I'm not against move that to a CSS variable.

It's weird, I couldn't see any reason that it shouldn't work but in the inspector the `min-width` (the result of the  `calc()`) was always `0`

**Pull request merged**